mkt/files/models.py

```diff
@@ -103,7 +105,7 @@ def addon(self):
     def generate_hash(self, filename=None):
         """Generate a hash for a file."""
         hash = hashlib.sha256()
-        with storage.open(filename or self.file_path, 'rb') as obj:
+        with private_storage.open(filename or self.file_path, 'rb') as obj:
```

Comment:
On the other hand, we should only call it once when creating the `File` instance. We re-use `File` instances for hosted apps, but I wonder if we really use their hash (or their `File` for that matter)